runtime/src/iree/hal/drivers/cuda/cuda_driver.c

```diff
@@ -48,6 +52,30 @@ IREE_API_EXPORT void iree_hal_cuda_driver_options_initialize(
   out_options->default_device_index = 0;
 }
 
+#if IREE_HAL_DRIVER_CUDA_NCCL
+
+static iree_status_t iree_hal_nccl_get_unique_id_from_env(
+    iree_hal_cuda_driver_t* driver) {
+  IREE_TRACE_ZONE_BEGIN(z0);
+
+  char* nccl_comm_id_str = getenv("NCCL_COMM_ID");
+  if (!nccl_comm_id_str) {
+    IREE_TRACE_ZONE_END(z0);
+    return iree_make_status(IREE_STATUS_INTERNAL);
```

Comment:
Always include error messages in status messages so that users can handle them ("expected NCCL_COMM_ID environment variable to be set when using the default NCCL configuration").
`INTERNAL` should be used only for internal errors that the user has no control over - here, the user didn't specify an argument via the environment and IREE_STATUS_INVALID_ARGUMENT would communicate that.